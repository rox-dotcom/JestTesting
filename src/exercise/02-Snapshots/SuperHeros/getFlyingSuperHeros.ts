interface ISuperHero {
  name: string;
  power: string[];
}

export function getFlyingSuperHeros(superHeros: ISuperHero[]): ISuperHero[] {
  return superHeros.filter((superHero) => superHero.power.includes("Fly"));
}
