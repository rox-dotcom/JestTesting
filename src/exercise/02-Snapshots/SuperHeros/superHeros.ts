export const superHeros = [
  { name: "Superman", power: ["Fly", "Super Strength"] },
  { name: "Hulk", power: ["Super Strength", "Regeneration"] },
  {
    name: "IronMan",
    power: ["Intelligence", "Technology", "Fly", "Billionaire"],
  },
  { name: "SpiderMan", power: ["Agility", "Spider-Sense"] },
  { name: "CaptainAmerica", power: ["Leadership", "Super Strength"] },
  { name: "BlackWidow", power: ["Agility", "Intelligence"] },
  { name: "GreenLantern", power: ["Energy Manipulation", "Fly"] },
  { name: "Wolverine", power: ["Regeneration", "Claws"] },
  { name: "Thor", power: ["Thunder Control", "Super Strength"] },
  { name: "Batman", power: ["Intelligence", "Martial Arts", "Billionaire"] },
  { name: "Flash", power: ["Super Speed", "Time Travel"] },
  { name: "WonderWoman", power: ["Super Strength", "Agility", "Fly"] },
  //{ name: "Aquaman", power: ["Water Control", "Super Strength"] },
];
