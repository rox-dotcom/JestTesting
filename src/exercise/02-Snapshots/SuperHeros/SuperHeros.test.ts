/* Exercise 2: Test using snapshots */

/* Mock the function using jest.fn().
Write three tests inside a describe block. You should use import the superHeros[] and getFlyingSuperHeros function.

1. First Test: 
Test should return an empty array if no superheros have the 'Fly' power"
2. Second Test:
Test should return an array of superHeros that have the 'Fly' power"
3. Third Test:
Test should match the snapshot of flying superheros.
The snapshot file should contain the expected output of the test.
The snapshot should be saved in a __snapshots__ directory next to the test file.
The snapshot file should be named SuperHeros.test.ts.snap.
*/

import { superHeros } from "./superHeros";
import { getFlyingSuperHeros } from "./getFlyingSuperHeros";


describe("getFlyingSuperHeros", () => {
  it("No superheros fly", () => {
   getFlyingSuperHeros([]);
   
 });

 it("Some superheros fly", () => {
   getFlyingSuperHeros(superHeros);
 });

 it("snapshot of flying superheros", () => {
  const snapshot = getFlyingSuperHeros(superHeros);
  expect(snapshot).toMatchSnapshot();
 })

})
