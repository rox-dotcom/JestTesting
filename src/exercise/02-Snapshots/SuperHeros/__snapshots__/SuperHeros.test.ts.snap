// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFlyingSuperHeros snapshot of flying superheros 1`] = `
[
  {
    "name": "Superman",
    "power": [
      "Fly",
      "Super Strength",
    ],
  },
  {
    "name": "IronMan",
    "power": [
      "Intelligence",
      "Technology",
      "Fly",
      "Billionaire",
    ],
  },
  {
    "name": "GreenLantern",
    "power": [
      "Energy Manipulation",
      "Fly",
    ],
  },
  {
    "name": "WonderWoman",
    "power": [
      "Super Strength",
      "Agility",
      "Fly",
    ],
  },
]
`;
